Felienne/hedy
# [CLEAN-UP] Simplified saving and sharing

**Description**
In this PR we clean-up the structure for saving and sharing programs. Currently we only verify the correctness of programs when sharing one. But, why should we? We simplify our code by always parsing the code on save, independent of sharing or not. This way we reduce the function calls as we can always call the `saveit` function, only with an additional value for sharing or not.

**Fixes**
This fixes nothing, but cleans-up.

**How to test**
Make sure you are logged in and verify that all saving and sharing functionality is still working.
